pkg/kubectl/cmd/apply.go

```diff
@@ -608,7 +607,10 @@ func (p *patcher) patchSimple(obj runtime.Object, modified []byte, source, names
 			return nil, nil, cmdutil.AddSourceToErr(fmt.Sprintf(createPatchErrFormat, original, modified, current), source, err)
 		}
 	}
-
+	if string(patch) == "{}" {
+		glog.Infof("no changes found, donot need to patch")
```

Comment:
glog isn't really used for user-facing messages in kubectl. I think rather than printing the success message, "pod blah configured", it should print a message like this.

I.e. line 344 needs to change:
https://github.com/kubernetes/kubernetes/blob/067f2db355db264e5b58cb5f166ba88cb07b2ed6/pkg/kubectl/cmd/apply.go#L344